go/vt/vttablet/tabletconn/grpc_error.go

```diff
@@ -49,3 +59,19 @@ func ErrorFromVTRPC(err *vtrpcpb.RPCError) error {
 	}
 	return vterrors.Errorf(err.Code, "vttablet: %s", err.Message)
 }
+
+func ErrorsFromVTRPCWrapTransientTxError(err *vtrpcpb.RPCError, inTx bool) error {
+	nerr := ErrorFromVTRPC(err)
+	if nerr == nil || !inTx {
+		return nerr
+	}
+
+	return ifConnectionRefusedWrapInVT15001(nerr)
+}
+
+func ifConnectionRefusedWrapInVT15001(err error) error {
+	if vterrors.Code(err) == vtrpcpb.Code_UNAVAILABLE && strings.Contains(err.Error(), "connection refused") {
+		return vterrors.VT15001(vtrpcpb.Code_UNAVAILABLE, err.Error())
+	}
+	return err
+}
```

Comment:
unit test will help here to check if all the 3 errors messages are handled

Reply:
At this level we only get the second error listed in the PR's description. The first error listed comes from VTTablet, the third comes from the healthcheck.

I added more tests anyway:  89060c80363

Reply:
I removed this methods via f5ee299c19f